File: apphub-lib/apphub-lib-config/apphub-lib-config-common/src/main/java/com/github/saphyra/apphub/lib/config/common/endpoints/UserEndpoints.java
package com.github.saphyra.apphub.lib.config.common.endpoints;

public class UserEndpoints {
    //Login Session
    public static final String EVENT_DELETE_EXPIRED_ACCESS_TOKENS = "/event/delete-expired-access-tokens";
    public static final String EVENT_REFRESH_ACCESS_TOKEN_EXPIRATION = "/event/refresh-access-token-expiration";
    public static final String LOGIN = "/api/user/authentication/login";
    public static final String LOGOUT = "/api/user/authentication/logout";
    public static final String CHECK_SESSION = "/api/user/authentication/session";
    public static final String EXTEND_SESSION = "/api/user/authentication/session";
    public static final String USER_DATA_INTERNAL_GET_ACCESS_TOKEN_BY_ID = "/internal/user/authentication/{accessTokenId}";
    public static final String EVENT_ACCESS_TOKEN_INVALIDATED = "/event/access-token-invalidated";

    //Account
    public static final String EVENT_TRIGGER_ACCOUNT_DELETION = "/event/user-data/trigger-account-deletion";
    public static final String ACCOUNT_REGISTER = "/api/user";
    public static final String ACCOUNT_CHANGE_LANGUAGE = "/api/user/account/language";
    public static final String ACCOUNT_CHANGE_EMAIL = "/api/user/account/email";
    public static final String ACCOUNT_CHANGE_USERNAME = "/api/user/account/username";
    public static final String ACCOUNT_CHANGE_PASSWORD = "/api/user/account/password";
    public static final String ACCOUNT_DELETE_ACCOUNT = "/api/user/account";
    public static final String USER_DATA_SEARCH_ACCOUNT = "/api/user/accounts";
    public static final String USER_DATA_GET_ACCOUNT = "/api/user/account";
    public static final String USER_DATA_INTERNAL_GET_USER_LANGUAGE = "/internal/user/{userId}/data/language";
    public static final String USER_DATA_INTERNAL_GET_ACCOUNT = "/internal/api/user/data/{userId}";
    public static final String USER_DATA_INTERNAL_USER_EXISTS = "/internal/api/user/data/{userId}/exists";
    public static final String USER_DATA_GET_USERNAME = "/api/user/data/name";

    //Ban
    public static final String EVENT_TRIGGER_REVOKE_EXPIRED_BANS = "/event/user-data/trigger-remove-expired-bans";
    public static final String ACCOUNT_BAN_USER = "/api/user/ban";
    public static final String ACCOUNT_REVOKE_BAN = "/api/user/ban/{banId}";
    public static final String ACCOUNT_GET_BANS = "/api/user/ban/{userId}";
    public static final String ACCOUNT_MARK_FOR_DELETION = "/api/user/ban/{userId}/mark-for-deletion";
    public static final String ACCOUNT_UNMARK_FOR_DELETION = "/api/user/ban/{userId}/mark-for-deletion";
    public static final String ACCOUNT_BAN_SEARCH = "/api/user/ban/search";
    public static final String ACCOUNT_BAN_GET_DETAILS_FOR_ERROR_PAGE = "/api/user/ban/details";

    //SETTINGS
    public static final String GET_USER_SETTINGS = "/api/user/settings/{category}";
    public static final String SET_USER_SETTINGS = "/api/user/settings";

    //Role
    public static final String USER_DATA_GET_USER_ROLES = "/api/user/data/roles";
    public static final String USER_DATA_ADD_ROLE = "/api/user/data/roles";
    public static final String USER_DATA_REMOVE_ROLE = "/api/user/data/roles";
    public static final String USER_DATA_DISABLE_ROLE = "/api/user/data/roles/{role}";
    public static final String USER_DATA_ENABLE_ROLE = "/api/user/data/roles/{role}";
    public static final String USER_DATA_GET_DISABLED_ROLES = "/api/user/data/roles/disabled";
    public static final String USER_DATA_ADD_ROLE_TO_ALL = "/api/user/data/roles/all/{role}";
    public static final String USER_DATA_REMOVE_ROLE_FROM_ALL = "/api/user/data/roles/all/{role}";
    public static final String USER_DATA_ROLES_FOR_ALL_RESTRICTED = "/api/user/data/roles/restricted";
}

File: apphub-frontend/src/common/js/dao/endpoints/UserEndpoints.js
import { Endpoint, RequestMethod } from "../dao";

export const USER_DATA_GET_USERNAME = new Endpoint(RequestMethod.GET, "/api/user/data/name");
export const ACCOUNT_CHANGE_LANGUAGE = new Endpoint(RequestMethod.POST, "/api/user/account/language");
export const ACCOUNT_CHANGE_EMAIL = new Endpoint(RequestMethod.POST, "/api/user/account/email");
export const ACCOUNT_CHANGE_USERNAME = new Endpoint(RequestMethod.POST, "/api/user/account/username");
export const ACCOUNT_CHANGE_PASSWORD = new Endpoint(RequestMethod.POST, "/api/user/account/password");
export const ACCOUNT_DELETE_ACCOUNT = new Endpoint(RequestMethod.DELETE, "/api/user/account");
export const ACCOUNT_GET_USER = new Endpoint(RequestMethod.GET, "/api/user/account");

export const ACCOUNT_REGISTER = new Endpoint(RequestMethod.POST, "/api/user");
export const LOGIN = new Endpoint(RequestMethod.POST, "/api/user/authentication/login");
export const LOGOUT = new Endpoint(RequestMethod.POST, "/api/user/authentication/logout");

export const GET_USER_SETTINGS = new Endpoint(RequestMethod.GET, "/api/user/settings/{category}");
export const SET_USER_SETTINGS = new Endpoint(RequestMethod.POST, "/api/user/settings");

export const USER_DATA_ROLES_FOR_ALL_RESTRICTED = new Endpoint(RequestMethod.GET, "/api/user/data/roles/restricted");
export const ADMIN_PANEL_AVAILABLE_ROLES = new Endpoint(RequestMethod.GET, "/api/user/data/roles/restricted");
export const USER_DATA_ADD_ROLE_TO_ALL = new Endpoint(RequestMethod.POST, "/api/user/data/roles/all/{role}");
export const USER_DATA_REMOVE_ROLE_FROM_ALL = new Endpoint(RequestMethod.DELETE, "/api/user/data/roles/all/{role}");

export const USER_DATA_GET_USER_ROLES = new Endpoint(RequestMethod.POST, "/api/user/data/roles");
export const USER_DATA_ADD_ROLE = new Endpoint(RequestMethod.PUT, "/api/user/data/roles");
export const USER_DATA_REMOVE_ROLE = new Endpoint(RequestMethod.DELETE, "/api/user/data/roles");
export const IS_ADMIN = new Endpoint(RequestMethod.GET, "/api/user/admin");

export const USER_DATA_GET_DISABLED_ROLES = new Endpoint(RequestMethod.GET, "/api/user/data/roles/disabled");
export const USER_DATA_ENABLE_ROLE = new Endpoint(RequestMethod.DELETE, "/api/user/data/roles/{role}");
export const USER_DATA_DISABLE_ROLE = new Endpoint(RequestMethod.PUT, "/api/user/data/roles/{role}");

export const ACCOUNT_BAN_SEARCH = new Endpoint(RequestMethod.POST, "/api/user/ban/search");
export const ADMIN_PANEL_BAN_DETAILS_PAGE = new Endpoint(RequestMethod.GET, "/web/admin-panel/ban/{userId}");
export const ACCOUNT_GET_BANS = new Endpoint(RequestMethod.GET, "/api/user/ban/{userId}");
export const ACCOUNT_MARK_FOR_DELETION = new Endpoint(RequestMethod.DELETE, "/api/user/ban/{userId}/mark-for-deletion");
export const ACCOUNT_UNMARK_FOR_DELETION = new Endpoint(RequestMethod.POST, "/api/user/ban/{userId}/mark-for-deletion");
export const ACCOUNT_BAN_USER = new Endpoint(RequestMethod.PUT, "/api/user/ban");
export const ACCOUNT_REVOKE_BAN = new Endpoint(RequestMethod.DELETE, "/api/user/ban/{banId}");
export const ACCOUNT_BAN_GET_DETAILS_FOR_ERROR_PAGE = new Endpoint(RequestMethod.POST, "/api/user/ban/details");